apps/expo/src/app/workout-completion/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { router, useLocalSearchParams } from 'expo-router'
+
+import { AntDesign, Ionicons } from '@expo/vector-icons'
+
+import RotatingBarbellIcon from '~/components/notif/RotatingBarbellIcon'
+import { TExercise } from '~/components/tracker/workout-tracker'
+import Button from '~/components/ui/button/button'
+import { useGlobalContext } from '~/context/global-context'
+import colors from '~/styles/colors'
+import { formatForWorkoutCompletion } from '~/utils/timerFormatter'
+import { api } from '~/utils/trpc/api'
+
+const WorkoutCompletion: React.FC = () => {
+  const { userData } = useGlobalContext()
+  const { data, isFetching } = api.user.getUserWorkoutHistory.useQuery({
+    userId: userData!.id,
+  })
+
+  const { workoutName, exercises, duration, dateFinished } = useLocalSearchParams() // TODO: type this
+  console.log('workoutName', workoutName)
+  console.log('exercises', exercises)
+  console.log('duration', duration)
+  const parsedExercises: TExercise[] = JSON.parse(exercises as string)
+
+  return (
+    <SafeAreaView style={{ backgroundColor: '#1E1E1E', flex: 1 }}>
+      <View className='h-full px-4'>
+        {!isFetching ? (
+          <>
+            <View className='flex items-start'>
+              <Button
+                color='dark'
+                className='flex p-2'
+                onPress={() => router.replace('(dashboard)/')}
+              >
+                <AntDesign name='close' size={30} color='white' />
+              </Button>
+            </View>
+
+            <View className='mb-4 flex flex-1 justify-between'>
+              <View>
+                <View className='flex flex-row items-center justify-center gap-x-4'>
+                  <Text className='text-3xl'>🎉</Text>
+                  <Text className='font-koulen text-5xl font-bold text-white'>WOOHOO!</Text>
+                </View>
+                <View className='mt-6 flex items-center justify-center'>
+                  <Text className='text-2xl font-bold text-white'>Workout Completed</Text>
+                  <Text className='mt-2 text-lg text-white'>{`You have completed ${data?.workoutHistory.length} workouts.`}</Text>
```

Comment:
Instead of the string interpolation you should be able to do:

```tsx
<Text ...>You have completed {data?.workoutHistory.length} workouts.</Text>
```

Also, I would add a null coalesce (`??`) to keep from showing "null" on the screen when `data` is null.